perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,161 @@
+        percentile_latency_90 FLOAT64, 
+        percentile_latency_95 FLOAT64, 
+        FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+      ) OPTIONS (description = 'Table for storing FIO metrics extracted from experiments.');
+    """.format(self.project_id, self.dataset_id, BigQuery.fio_table_id, self.dataset_id, BigQuery.configuration_table_id)
+
+    # Query for creating vm_metrics table
+    query_create_table_vm_metrics = """
+      CREATE TABLE IF NOT EXISTS {}.{}.{}(
+        configuration_id INT64, 
+        start_time_build TIMESTAMP,
+        end_time INT64, 
+        cpu_utilization_peak_percentage FLOAT64, 
+        cpu_utilization_mean_percentage FLOAT64, 
+        received_bytes_peak_per_sec FLOAT64, 
+        received_bytes_mean_per_sec FLOAT64, 
+        read_bytes_count INT64,
+        ops_error_count INT64, 
+        ops_mean_latency_sec FLOAT64, 
+        sent_bytes_peak_per_sec FLOAT64, 
+        sent_bytes_mean_per_sec FLOAT64, 
+        sent_bytes_count INT64,
+        memory_utilization_ram FLOAT64,
+        memory_utilization_disk_tempdir FLOAT64,
+        iops FLOAT64, 
+        ops_count_list_object INT64, 
+        ops_count_create_object INT64, 
+        ops_count_stat_object INT64, 
+        ops_count_new_reader INT64, 
+        FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+      ) OPTIONS (description = 'Table for storing VM metrics extracted from experiments.');
+    """.format(self.project_id, self.dataset_id, BigQuery.vm_table_id, self.dataset_id, BigQuery.configuration_table_id)
+
+    # Query for creating ls_metrics table
+    query_create_table_ls_metrics = """
+      CREATE TABLE IF NOT EXISTS {}.{}.{}(
+        configuration_id INT64,
+        start_time_build TIMESTAMP,
+        mount_type STRING, 
+        command STRING,
+        start_time FLOAT64, 
+        end_time FLOAT64,
+        num_files INT64, 
+        num_samples INT64, 
+        min_latency_msec FLOAT64,
+        max_latency_msec FLOAT64,
+        mean_latency_msec FLOAT64, 
+        median_latency_msec FLOAT64, 
+        standard_dev_latency_msec FLOAT64, 
+        percentile_latency_20 FLOAT64, 
+        percentile_latency_50 FLOAT64, 
+        percentile_latency_90 FLOAT64, 
+        percentile_latency_95 FLOAT64, 
+        cpu_utilization_peak_percentage FLOAT64, 
+        cpu_utilization_mean_percentage FLOAT64,            
+        received_bytes_peak_per_sec FLOAT64, 
+        received_bytes_mean_per_sec FLOAT64,
+        memory_utilization_ram FLOAT64, 
+        FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+      ) OPTIONS (description = 'Table for storing GCSFUSE metrics extracted from list experiments.');
+    """.format(self.project_id, self.dataset_id, BigQuery.ls_table_id, self.dataset_id, BigQuery.configuration_table_id)
+
+    self._execute_query_and_check_error(query_create_table_experiment_configuration)
```

Comment:
check for transactions here.

Reply:
@RuchikaSharma-8 mentioned that transaction is not possible while creating tables in a dataset.